axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveValidateServiceImpl.java

```diff
@@ -203,6 +204,23 @@ public void checkPreconditions(Move move) throws AxelorException {
               String.format(I18n.get(IExceptionMessage.MOVE_11), moveLine.getName()));
         }
 
+        if (move.getFunctionalOriginSelect() == null
+            || !(journal
+                .getAuthorizedFunctionalOriginSelect()
+                .contains(move.getFunctionalOriginSelect().toString()))) {
+          throw new AxelorException(
+              TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+              String.format(
+                  I18n.get(IExceptionMessage.MOVE_13),
+                  MetaStore.getSelectionItem(
+                          "iaccount.move.functional.origin.select",
+                          move.getFunctionalOriginSelect().toString())
```

Comment:
NPE will be thrown here
Use other msg,when move.functionalOriginSelect is null
Also when a function call has many args then avoid adding another func call in it  to improve code readability instead store in variable